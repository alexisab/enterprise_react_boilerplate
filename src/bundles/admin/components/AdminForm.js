import React from 'react';
import { Button, Form, Input, Icon } from 'antd';
import enhance from '../composers/AdminFormComposer';

const FormItem = Form.Item;

const AdminForm = (props) => {
  const { formData, setFormData, formErrors, handleSave, handleClose } = props;

  return (
    <Form layout="vertical">
      <FormItem
        label="Name"
        validateStatus={formErrors.name && 'error'}
        hasFeedback
        help={formErrors.name}
      >
        <Input
          prefix={<Icon type="user" style={{ fontSize: 13 }} />}
          placeholder="Name"
          onChange={e => setFormData({ ...formData, name: e.target.value })}
        />
      </FormItem>
      <FormItem
        label="Email"
        validateStatus={formErrors.email && 'error'}
        hasFeedback
        help={formErrors.email}
      >
        <Input
          prefix={<Icon type="mail" style={{ fontSize: 13 }} />}
          placeholder="Email"
          onChange={e => setFormData({ ...formData, email: e.target.value })}
        />
      </FormItem>
      <div className="dialog-form-btn-grp">
        <Button size="large" onClick={handleClose}>Cancel</Button>
        <Button type="primary" className="save-btn" size="large" onClick={handleSave}>Create Admin</Button>
      </div>
    </Form>
  );
}

export default enhance(AdminForm);
