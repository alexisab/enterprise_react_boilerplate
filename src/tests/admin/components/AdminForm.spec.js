import React from "react";
import renderer from "react-test-renderer";
// import { shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import AdminForm from "../../../bundles/admin/components/AdminForm";

const handleSave = jest.fn();
const setFormData = jest.fn();
const handleClose = jest.fn();

const props = {
  handleSave,
  handleClose,
  setFormData,
  formData: {
    name: "",
    email: ""
  },
  formErrors: {
    name: "",
    email: ""
  }
};

describe("AdminForm", () => {
  test("it should match snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <AdminForm {...props} />
      </MemoryRouter>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
