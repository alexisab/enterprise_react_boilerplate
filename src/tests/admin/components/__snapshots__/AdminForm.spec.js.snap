// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminForm it should match snapshot 1`] = `
<form
  className="ant-form ant-form-vertical"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={undefined}
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper"
            style={undefined}
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="anticon anticon-user"
                style={
                  Object {
                    "fontSize": 13,
                  }
                }
              />
            </span>
            <input
              className="ant-input"
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Name"
              style={null}
              type="text"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={undefined}
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Email"
      >
        Email
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper"
            style={undefined}
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="anticon anticon-mail"
                style={
                  Object {
                    "fontSize": 13,
                  }
                }
              />
            </span>
            <input
              className="ant-input"
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Email"
              style={null}
              type="text"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="dialog-form-btn-grp"
  >
    <button
      className="ant-btn ant-btn-lg"
      onClick={[Function]}
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      className="ant-btn save-btn ant-btn-primary ant-btn-lg"
      onClick={[Function]}
      type="button"
    >
      <span>
        Create Admin
      </span>
    </button>
  </div>
</form>
`;
